src/features/security-cloud-game/SecurityCloudGame.js

```diff
@@ -1,82 +1,113 @@
-        /** Height to set the game to */
-        height: PropTypes.number.isRequired,
-        /** Function to call when the `DragCloud` is successfuly dropped onto the `DropCloud` */
-        onValidDrop: PropTypes.func,
-        /** Width to set the game to */
-        width: PropTypes.number.isRequired,
-    };
+// pick these numbers to balance accessibility and game complexity
+const CLOUD_SIZE_RATIO = 5;
+const GRID_TRACK_SIZE_RATIO = 20;
+
+const SecurityCloudGame = ({ height, intl: { formatMessage }, onValidDrop, width }) => {
+    const [dropCloudPosition, setDropCloudPosition] = useState(null);
+    const [dragCloudPosition, setDragCloudPosition] = useState(null);
+    const [layout, setLayout] = useState({});
+    const [gameState, setGameState] = useState({
+        isValidDrop: false,
+        isOverlap: false,
+        liveText: '',
+    });
+
+    const messageElementRef = useRef();
+    // to handle resize events
+    const gameBoardSizeRef = useRef({});
+
+    const { cloudSize, gameBoardHeight, gridTrackSize } = layout;
+    const { liveText, isOverlap, isValidDrop } = gameState;
+
+    useLayoutEffect(() => {
+        const { current: messageElement } = messageElementRef;
+        const newGameBoardHeight = height - messageElement.getBoundingClientRect().height;
+        // guardrail to prevent further rendering if the game board height is not positive
+        if (newGameBoardHeight <= 0) return;
+
+        const minGameBoardLength = Math.min(newGameBoardHeight, width);
+        setLayout({
+            gameBoardHeight: newGameBoardHeight,
+            cloudSize: minGameBoardLength / CLOUD_SIZE_RATIO,
+            gridTrackSize: minGameBoardLength / GRID_TRACK_SIZE_RATIO,
+        });
 
-    static defaultProps = {
-        cloudSize: 64,
-    };
+        messageElement.focus();
+    }, [height, width]);
 
-    constructor(props) {
-        super(props);
-        this.occupiedRegions = [];
-        this.state = {
-            isValidDrop: false,
-            isOverlap: false,
-        };
-
-        this.onDrag = throttle(this.onDrag, 100, {
-            leading: true,
-            trailing: true,
+    useEffect(() => {
+        if (!gameBoardHeight) {
+            return;
+        }
+
+        const { height: prevHeight, width: prevWidth } = gameBoardSizeRef.current;
+        const heightRatio = prevHeight ? gameBoardHeight / prevHeight : 1;
+        const widthRatio = prevWidth ? width / prevWidth : 1;
+
+        // declare and update this variable first in order to generate the starting position for drag cloud
+        let newDropCloudPosition;
+        // use prevState => {} to avoid referencing and updating the state at the same time
+        setDropCloudPosition(prevPos => {
+            newDropCloudPosition = !prevPos
+                ? getRandomCloudPosition(cloudSize, gameBoardHeight, width) // initial render
+                : { x: prevPos.x * widthRatio, y: prevPos.y * heightRatio }; // on board resize
```

Comment:
switching these might make it easier for the developer to read in their head:
e.g. "if there is a previous position, use the previous position"
vs. "if there is no previous position, get random cloud position"

```suggestion
            newDropCloudPosition = prevPos
                ? { x: prevPos.x * widthRatio, y: prevPos.y * heightRatio }; // on board resize
                : getRandomCloudPosition(cloudSize, gameBoardHeight, width) // initial render
```